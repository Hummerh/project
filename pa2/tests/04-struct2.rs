struct Point { x: i32, y: i32, z: i32 }

struct Dog {
    loc : Point,
    tailless: bool
}

fn main() {
    let p : Point = Point { x: 1, y: 2, z: 3};
    let d : Dog = Dog { loc: p, tailless: false };
}
