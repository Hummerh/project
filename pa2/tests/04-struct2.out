(crate
    (items
        (struct-def
            (id
                (Point))
            (field-defs
                (field-def
                    (id
                        (x))
                    (type-i32))
                (field-def
                    (id
                        (y))
                    (type-i32))
                (field-def
                    (id
                        (z))
                    (type-i32))))
        (struct-def
            (id
                (Dog))
            (field-defs
                (field-def
                    (id
                        (loc))
                    (id
                        (Point)))
                (field-def
                    (id
                        (tailless))
                    (type-bool))))
        (fn-def
            (id
                (main))
            (block
                (let
                    (pat-id
                        (id
                            (p)))
                    (id
                        (Point))
                    (struct
                        (id
                            (Point))
                        (field-inits
                            (field-init
                                (id
                                    (x))
                                (lit-dec))
                            (field-init
                                (id
                                    (y))
                                (lit-dec))
                            (field-init
                                (id
                                    (z))
                                (lit-dec)))))
                (let
                    (pat-id
                        (id
                            (d)))
                    (id
                        (Dog))
                    (struct
                        (id
                            (Dog))
                        (field-inits
                            (field-init
                                (id
                                    (loc))
                                (id
                                    (p)))
                            (field-init
                                (id
                                    (tailless))
                                (false)))))))))